src/main/java/org/jenkinsci/plugins/workflow/steps/ToolStep.java

```diff
@@ -119,7 +126,7 @@ public ListBoxModel doFillNameItems(@QueryParameter String type) {
             String name = step.getName();
             String type = step.getType();
             for (ToolDescriptor<?> desc : ToolInstallation.all()) {
-                if (type != null && !desc.getId().equals(type)) {
+                if (type != null && !desc.getId().equals(type) && !type.equals(SymbolLookup.getSymbolValue(desc))) {
```

Comment:
Good idea - I've verified it works both against 1.x and 2.x with Maven (i.e., without and with a symbol) but having an explicit test in there makes sense. I'll do that shortly.